docs/source-pytorch/advanced/model_parallel.rst

```diff
@@ -114,9 +118,37 @@ To reach larger parameter sizes and to be memory efficient, we have to shard par
     This is a limitation of Fully Sharded Training that will be resolved in the future.
 
 Enabling Module Sharding for Maximum Memory Efficiency
-""""""""""""""""""""""""""""""""""""""""""""""""""""""
+------------------------------------------------------
+
+Auto Wrapping
+^^^^^^^^^^^^^
+
+Model layers should be wrapped in FSDP in a nested way to save peak memory and enable communication and computation overlapping. The
+simplest way to do it is auto wrapping, which can serve as a drop-in replacement for DDP without changing the rest of the code. You don't
+have to ``wrap`` layers manually as in the case of manual wrapping.
+
+.. note::
+    While initializing the optimizers inside ``configure_optimizers`` hook, make sure to use ``self.trainer.model.parameters()``, else
+    PyTorch will raise an error. This is required because when you use auto-wrap, the model layers are sharded and your
+    ``lightning_module.parameters()`` will return a generator with no params.
```

Comment:
Oh no, this is a bummer. I think we should find a way to work around this. Maybe write: "This inconvenience will be addressed in the future".